recipes/capnproto/all/conanfile.py

```diff
@@ -0,0 +1,183 @@
+from conans import ConanFile, CMake, tools, AutoToolsBuildEnvironment
+from conans.errors import ConanInvalidConfiguration
+import glob
+import os
+
+
+class CapnprotoConan(ConanFile):
+    name = "capnproto"
+    description = "Cap'n Proto serialization/RPC system."
+    license = "MIT"
+    topics = ("conan", "capnproto", "serialization", "rpc")
+    homepage = "https://capnproto.org"
+    url = "https://github.com/conan-io/conan-center-index"
+    exports_sources = ("CMakeLists.txt", "patches/**")
+    generators = "cmake"
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "with_openssl": [True, False],
+        "with_zlib": [True, False]
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "with_openssl": True,
+        "with_zlib": True
+    }
+
+    _cmake = None
+    _autotools = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    @property
+    def _minimum_compilers_version(self):
+        return {
+            "Visual Studio": "15",
+            "gcc": "5",
+            "clang": "5",
+            "apple-clang": "5.1",
+        }
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+            del self.options.with_openssl
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+        if self.settings.compiler.cppstd:
+            tools.check_min_cppstd(self, 14)
+        minimum_version = self._minimum_compilers_version.get(str(self.settings.compiler), False)
+        if not minimum_version:
+            self.output.warn("Cap'n Proto requires C++14. Your compiler is unknown. Assuming it supports C++14.")
+        elif tools.Version(self.settings.compiler.version) < minimum_version:
+            raise ConanInvalidConfiguration("Cap'n Proto requires C++14, which your compiler does not support.")
+        if self.settings.compiler == "Visual Studio" and self.options.shared:
+            raise ConanInvalidConfiguration("Cap'n Proto doesn't support shared libraries for Visual Studio")
+
+    def requirements(self):
+        if self.options.get_safe("with_openssl"):
+            self.requires("openssl/1.1.1h")
+        if self.options.with_zlib:
+            self.requires("zlib/1.2.11")
+
+    def build_requirements(self):
+        if self.settings.os != "Windows":
```

Comment:
```suggestion
        if not tools.os_info.is_windows:
```

Reply:
No, it's really tied to the condition in `build()`.

Reply:
Oh yeah, you're correct.